org.eclipse.xtext/src/org/eclipse/xtext/resource/IResourceDescriptions.java

```diff
@@ -43,6 +44,16 @@
 
 		void setContext(Notifier ctx);
 	}
+	
+	/**
+	 * A resource set aware instance {@link IResourceDescriptions}.
+	 * 
+	 * @since 2.17
+	 */
+	interface IResourceSetAware extends IResourceDescriptions {
+
+		ResourceSet getResourceSet();
```

Comment:
What do you think about a default method in IResourceDesriptions, e.g `default Optional<ResourceSet> getContextResourceSet() { returns Optional.absent() }` with a clarifying JavaDoc, that describes what the context resource set is and under which circumstances it is present or absent?

Reply:
For existing impls that do have a getResourceSet, the getContextResourceSet should delegate to that one.

Reply:
If using default, what is the reason for not doing a default getResourceSet() {  return null; } with similar javadoc? Just trying to understand the difference. :-)

Reply:
Three things come to my mind:
- `getContextResourceSet` carries a slightly better semantics
- `Optional<ResourceSet>` makes it very explicit that this may be not available. 
- existing implementation may already have a method `getResourceSet` and it could potentially be undesirable to expose that one.

Reply:
Thanks!
Should I change to using Optional<> et al. or is the current solution of an additional interfance easier to understand?

Reply:
Let's keep the approach as is. The alternative with the default method is not a substantial improvement. Thank you for proving the patch!

Reply:
Np, thank you!